nireports/assembler/reportlet.py

```diff
@@ -270,6 +270,26 @@ def __init__(self, layout, config=None, out_dir=None, bids_filters=None, metadat
                         name=html_anchor,
                         style="; ".join(f"{k}: {v}" for k, v in style.items()),
                     )
+                elif ext in (".png", ".jpg", ".jpeg"):
+                    entities = dict(bidsfile.entities)
+                    if desc_text:
+                        desc_text = desc_text.format(**entities)
+
+                    try:
+                        html_anchor = src.relative_to(out_dir)
+                    except ValueError:
+                        html_anchor = src.relative_to(Path(layout.root))
+                        dst = out_dir / html_anchor
+                        dst.parent.mkdir(parents=True, exist_ok=True)
+                        copyfile(src, dst, copy=True, use_hardlink=True)
+
+                    style = {"width": "100%"} if is_static else {}
+                    style.update(config.get("style", {}))
+
+                    stylestr = "; ".join(f"{k}: {v}" for k, v in style.items())
+                    contents = f'<img src="{html_anchor}" style="{stylestr}"/>'
```

Comment:
I think we want to reuse the templates we use with SVG, e.g.:

```suggestion
                    contents = SVG_SNIPPET[1].format(name=html_anchor, style=stylestr)'
```

We might want to change it to `IMG_SNIPPET` and switch L269 to be `(IMG_SNIPPET if is_static else SVG_SNIPPET)`, and also update the CSS to use an `img-reportlet` class instead of `svg-reportlet`.

If these are hairy changes, I'm also okay with doing a minimal `IMG_SNIPPET` and not touching `SVG_SNIPPET`.

Reply:
I'm happy to abstract out the shared bits. I'll do that ASAP.